### PR TITLE
Download dcap from the archive if no git repository is found

### LINKED ISSUE #390: SGX 2.5 release `make` fails due to git module command
From the `Makefile` of SDK 2.5,
```
all: dcap_source sdk psw

# dcap_source:
# 	git submodule update --init --recursive
```
Simply running `make` from the source code downloaded from [here](https://github.com/intel/linux-sgx/releases/tag/sgx_2.5) won't work due to `git submodule` command

I had to manually download the source code for DataCenterAttestationPrimitives.

IMHO It would be good if the release source tar bells download source codes from the archive.

### PR DESCRIPTION
**Pull request by shpark**
This closes #390

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,6 +78,7 @@ clean:
 	@$(MAKE) -C sdk/                                clean
 	@$(MAKE) -C psw/                                clean
 	@$(RM)   -r $(ROOT_DIR)/build
+	@$(RM)   -rf external/dcap_source
```
Comment (llly) [inline Makefile]: `external/dcap_source` folder is part of the repo. Unstaged changes exist after `make clean`. Only remove  the contents inside the folder is enough.

Comment (shpark) [reply Makefile]: I just removed this line. Simply calling `rm -rf external/dcap_source/*` leaves `.git` behinds. Maybe we need something like `make -C external/dcap_source clean`?

File: Makefile [hunk 2]
```diff
@@ -36,7 +38,14 @@ include buildenv.mk
 all: dcap_source sdk psw
 
 dcap_source:
+ifeq ($(shell git rev-parse --is-inside-work-tree), true)
 	git submodule update --init --recursive
+else
+	curl --output dcap_source.tar.gz -L --tlsv1 ${DCAP_DOWNLOAD_BASE}/DCAP_${DCAP_VER}.tar.gz
```
Comment (thiagomacieira) [inline Makefile]: DO NOT download during Make.

Please add this to the script that downloads dependencies prior to build. instead.

Comment (shpark) [reply Makefile]: Sorry for the late response, I've missed to answer to your comment. Then, should this change be reverted?

**Pull request merged**